week4/src/component/common/Button/Button.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { Theme } from '../../../styles/theme';
+
+interface ButtonProps {
+  text: string;
+  onClick?: () => void;
+  hoverColor?: string;
+  disabled?: boolean;
+}
+
+const StyledButton = styled.button<{ hoverColor?: string }>`
+  background-color: ${Theme.color.buttonColor};
+  padding: 0.5rem 1rem;
+  border: none;
+  cursor: pointer;
+  font-size: 1rem;
+  width: 25rem;
+  height: 3rem;
+  transition: background-color 0.3s ease, color 0.3s ease;
+
+  &:hover {
+    ${({ hoverColor }) =>
+      hoverColor
+        ? css`
+            background-color: ${hoverColor};
+          `
+        : css`
+            background-color: #0056b3;
+          `}
+  }
```

Comment:
이 부분은 css를 따로 작성하는 것보다 한줄로 작성하는 것이 더 직관적이고 코드도 간결할 것 같습니다.
```suggestion
  &:hover {
    background-color: ${({ hoverColor }) => hoverColor || "#0056b3"};
  }
```